cmd/ddev/cmd/dotenv-set.go

```diff
@@ -101,7 +101,7 @@ ddev dotenv set .ddev/.env.redis --redis-tag 7-bookworm`,
 		// Prepare the friendly message with formatted environment variables
 		var formattedVars []string
 		for _, k := range rawResultKeys {
-			formattedVars = append(formattedVars, fmt.Sprintf(`%s="%v"`, k, strings.ReplaceAll(changedEnvMap[k], `"`, `\"`)))
+			formattedVars = append(formattedVars, fmt.Sprintf(`%s="%v"`, k, strings.ReplaceAll(strings.ReplaceAll(changedEnvMap[k], `$`, `\$`), `"`, `\"`)))
```

Comment:
Wow, that's obscure! Needs a comment for sure.  But looks to me like this might be better as two steps rather than nested `strings.ReplaceAll()`, then it would be clearer. Can be done and assigned to a variable, then used.

Reply:
Extracted this logic to `ddevapp.EscapeEnvFileValue()`.